syntax = "proto3";

service BlogPoster {
  // Creates a blog post
  rpc CreatePost (CreateBlogPost) returns (BlogPost) {}
  rpc GetPost (GetPostRequest) returns (BlogPost) {}
  rpc GetAllPost (GetAllPostRequest) returns (GetAllPostResponse) {}
}
message BlogPost {
  int32  id_ = 1;
  string title = 2;
  string text = 3;
  bool published = 4;
}
message CreateBlogPost {
  string title = 2;
  string text = 3;
  bool published = 4;
}

message GetPostRequest {
  int32  id_ = 1;
}

message GetAllPostRequest {
  int32  page = 1;
  int32  per_page = 2;
}

message GetAllPostResponse {
  int32  page = 1;
  int32  per_page = 2;
  repeated BlogPost records = 3;
}
